ui/src/services/QueryService.ts

```diff
@@ -51,6 +51,17 @@ const QueryService = {
     });
   },
 
+  async getDisplayFromQueryIri(iri: string, includeLogicDesc: boolean): Promise<Query> {
+    return axios.get(Env.API + "api/query/public/queryDisplay", { params: { queryIri: iri, includeLogicDesc: includeLogicDesc } });
+  },
+
+  async getDisplayFromQuery(query: Query, includeLogicDesc: boolean): Promise<Query> {
+    return axios.post(Env.API + "api/query/public/queryDisplayFromQuery", query, {
+      params: { includeLogicDesc: includeLogicDesc }
```

Comment:
Can't combine POST body and request params. IMAPI will ignore request params. Create a new request model.